### DISCUSSION TITLE
Open html attachments if available

### DISCUSSION THREAD

**Discussion by wgmayer0** [neomutt/neomutt#4509] [Development]
The following code is in my `muttrc`:
`macro index,pager l "<view-attachments><search>html<enter><view-mailcap><exit>"`

How would I have mutt do the following?
1. Check if html is in any of the attachments
2. If it is, open it and quit any other actions
   If it is not, search for text/plain and open it

Right now, all it can do is open html documents but it fails if there are no html documents attached.

**Comment by mschilli87** [comment]
context: follow-up to https://github.com/neomutt/neomutt/issues/3168

**Comment by dcpurton** [comment]
What happens at the moment for you if there's no `text/html` part?

For me it just runs `<view-mailcap>` on the top attachment in the email and if there's no `text/html` part this is typically a `text/plain` part, which just displays as text in a pager.

**Comment by wgmayer0** [comment]
![1](https://user-images.githubusercontent.com/73801421/150071913-c97fc3e6-db41-4b28-9177-50c1398e8ee6.png)
![2](https://user-images.githubusercontent.com/73801421/150071920-8f6d083f-4eeb-446e-8ba8-5885c4bb31c5.png)
It will show the attachments for about 1 second then back out to the first screen

**Comment by dcpurton** [comment]
Ah! It has a `multipart/alternative` attachment but with only one alternative. IMHO that's weird.

What about:

```
macro index,pager l "<view-attachments><search>text/plain<enter><search>text/html<enter><view-mailcap><exit>"
```

And make sure you have a suitable line for `text/plain` in you `~/.mailcap` file.

**Comment by wgmayer0** [comment]
Same behavior as the original code. I actually already tried that
believe it or not.

**Comment by dcpurton** [comment]
What happens if you run `<view-mailcap>` on a `text/plain` part?

**Comment by wgmayer0** [comment]
When I am in attachments view, typing `:view-mailcap` while a `plain/text` is selected says `unknown command`

**Comment by dcpurton** [comment]
Perhaps it's already bound to `m`?

Otherwise, I notice that running commands by typing `:view-mailcap` doesn't work. So bind it to something in your `~/.neomuttrc`:

```
bind attach m view-mailcap
```

**Comment by wgmayer0** [comment]
Ok I did that, ran it, and got `mailcap entry for type text/plain not found`

**Comment by mschilli87** [comment]
@wgmayer0:
> Ok I did that, ran it, and got `mailcap entry for type text/plain not found`

Sounds like you missed @dcpurton earlier comment:
> And make sure you have a suitable line for text/plain in you ~/.mailcap file.

:wink:

**Comment by wgmayer0** [comment]
After fixing my mailcap file, this works. It is a little slow is the only issue.

**Comment by dcpurton** [comment]
Qutebrowser was new to me. But I find opening that to be the slow step. The macro and viewing the text part is quick.

**Comment by wgmayer0** [comment]
Thanks yes qutebrowser does take a while but I'm actually referring to less
for text/plain which is taking as long as qutebrowser

**Comment by dcpurton** [comment]
> Thanks yes qutebrowser does take a while but I'm actually referring to less for text/plain which is taking as long as qutebrowser

What about using `cat` then?

```
text/plain; cat %s; copiousoutput
```

**Comment by wgmayer0** [comment]
cat gives me the same delay it feels like

**Comment by wgmayer0** [comment]
If I pipe to less, it opens instantly. Same with cat

**Comment by dcpurton** [comment]
Sorry, I don't know what is wrong for you. If I `<view-mailcap>` a `text/plain` attachment it's instant for me.

**Comment by wgmayer0** [comment]
Interesting. It works. just not sure what is creating the delay

**Comment by wgmayer0** [comment]
How can I use this on multiple tagged messages? It takes qutebrowser a little bit of time to load so how can I have them all load at the same time so I can blast through the emails and they will all be ready?
